### ISSUE TITLE
It seems something went wrong while linking. Error: buildSettings.LIBRARY_SEARCH_PATHS.push

### ISSUE THREAD

**Issue by alexleventer** [rnpm/rnpm#168]
└── react-native-fbsdk@0.1.0 

rnpm-link info Linking react-native-fbsdk ios dependency 
rnpm-link ERR! It seems something went wrong while linking. Error: buildSettings.LIBRARY_SEARCH_PATHS.push is not a function 
Please file an issue here: https://github.com/rnpm/rnpm/issues 
rnpm-link info Module react-native-fbsdk has been successfully installed & linked

**Comment by alexleventer** [comment]
v1.7.0

**Comment by Kureev** [comment]
cc @grabbou

**Comment by grabbou** [comment]
Interesting, as always in that case, mind sending me your `project.pbxproj` from inside your `.xcodeproj` via email? It doesn't contain any sensitive information (like keys and passwords) so I can use it as a reproduction.

Fix is easy, but need to understand `why` it happens as that's originating from `node-xcode` lib.

**Comment by bsy** [comment]
running into the same issue with `react-native-fbsdk@0.2.0`, any update on this yet @grabbou ?

**Comment by alexleventer** [comment]
My issue was w/ `react-native-fbsdk@0.2.0` too

**Comment by grabbou** [comment]
Yup, I've received the `pbxproj` and I am working on a fix today.

**Comment by grabbou** [comment]
Turns out the `pbxproj` I've received belongs to other issue request, anyone from this topic mind sending it over to me? My email is listed on my profile, or just add `@gmail.com` to my handle.

**Comment by alexleventer** [comment]
Sorry for the delay. Sent @grabbou

**Comment by grabbou** [comment]
Thanks, now I've finally tracked down the problem to this line: https://github.com/alunny/node-xcode/blob/master/lib/pbxProject.js#L1146

It only checks if the value is `$(inherited)`. In case of your `pbxproj`, it's other string value which makes it break. I'll send a patch to `node-xcode` to work with that case tomorrow and hopefully we ship it soon

**Comment by alexleventer** [comment]
Sweet thanks @grabbou

**Comment by grabbou** [comment]
PS. If you can just prepend all your in-project `library search paths` with `$(inherited)`, that should fix the issue temporarily.

**Comment by alexleventer** [comment]
@grabbou what do you mean by in-project?

**Comment by alexleventer** [comment]
I'm running into the following issue when I prepend `$(inhertied)`: ```ld: 773 duplicate symbols for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)```

**Comment by grabbou** [comment]
Fix submitted to node-xcode: https://github.com/alunny/node-xcode/pull/96

**Comment by npomfret** [comment]
I've got the same error with 1.9.0.   Assuming it's going through my package.json in order its failing to link `react-native-device-info`.

 Whats the fix please?

**Comment by grabbou** [comment]
We can't seem to find a compromise with the library author on how the 96 PR (linked above) should be tested. We are investigating forking it temporarily and provide some additional support. If we don't come up with a solution by EOW, we will have to do it I believe.

**Comment by Kureev** [comment]
What's the status of this issue? Can we find a next step for it?
